api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationEffectiveDatetimeTranslatorImpl.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.Type;
+import org.openmrs.Obs;
+import org.openmrs.module.fhir2.api.translators.ObservationEffectiveDatetimeTranslator;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class ObservationEffectiveDatetimeTranslatorImpl implements ObservationEffectiveDatetimeTranslator {
+	
+	@Override
+	public Type toFhirResource(Obs obs) {
+		if (obs == null) {
+			return null;
+		}
+		
+		if (obs.getObsDatetime() != null) {
+			return (new DateTimeType(obs.getObsDatetime()));
+		}
+		
+		return null;
+	}
+	
+	@Override
+	public Obs toOpenmrsType(Obs obs, Type resource) {
+		if (resource == null) {
+			return obs;
+		}
+		
+		if (resource instanceof DateTimeType) {
```

Comment:
In addition to `DateTimeType` we should also support `InstantType` here, which is very similar to `DateTimeType`, but represents a slightly more precise type.